File: backend/app/models/goals.py
from datetime import date
from typing import Literal, Optional

from pydantic import BaseModel


class GoalItem(BaseModel):
    specific: str
    kpi_name: str
    kpi_target_value: float
    kpi_unit: str
    achievable: str
    timebound: date
    priority: Literal["Высокий", "Средний", "Низкий"]
    orientation_ids: list[str] = []
    project_names: list[str] = []
    orientation_explanation: Optional[str] = None


class ProjectCoverage(BaseModel):
    degree: Literal["Слабое", "Умеренное", "Сильное"]
    explanation: str


class RealismCheck(BaseModel):
    degree: Literal["Низкая", "Средняя", "Высокая"]
    score: int
    issues: list[str] = []


class GoalResponseItem(GoalItem):
    id: str
    context: Literal["USER_CREATED", "AI_SUGGESTED"]
    ai_explanation: Optional[str] = None
    project_coverage: Optional[ProjectCoverage] = None
    realism_check: Optional[RealismCheck] = None


class GoalsSubmitRequest(BaseModel):
    goals: list[GoalItem]


class GoalsResponse(BaseModel):
    goals: list[GoalResponseItem]


class GoalSuggestionsResponse(BaseModel):
    ai_suggestions: list[GoalResponseItem]

File: frontend/src/app/api/goals.ts
import { api } from './client';

export type GoalItem = {
  specific: string;
  kpi_name: string;
  kpi_target_value: number;
  kpi_unit: string;
  achievable: string;
  timebound: string;
  priority: 'Высокий' | 'Средний' | 'Низкий';
  orientation_ids?: string[];
  project_names?: string[];
  orientation_explanation?: string;
};

export type GoalResponseItem = GoalItem & {
  id: string;
  context: 'USER_CREATED' | 'AI_SUGGESTED';
  ai_explanation?: string;
  project_coverage?: { degree: string; explanation: string };
  realism_check?: { degree: string; score: number; issues: string[] };
};

export const goalsApi = {
  list: (contextId: string) =>
    api.get<{ goals: GoalResponseItem[] }>(`/contexts/${contextId}/goals`),
  submit: (contextId: string, goals: GoalItem[]) =>
    api.post<{ goals: GoalResponseItem[] }>(`/contexts/${contextId}/goals`, { goals }),
  suggestions: (contextId: string) =>
    api.get<{ ai_suggestions: GoalResponseItem[] }>(`/contexts/${contextId}/goals/suggestions`),
};
